tinygrad/tensor.py

```diff
@@ -191,13 +191,13 @@ def _data(self) -> memoryview:
   def data(self) -> memoryview:
     assert self.dtype.fmt is not None, f"no fmt dtype for {self.dtype}"
     assert all_int(self.shape), f"no data if shape is symbolic, {self.shape=}"
-    return self._data().cast(self.dtype.fmt, self.shape if len(self.shape) else (1,))
+    return self._data().cast(self.dtype.fmt, self.shape)
```

Comment:
a little unsure of this change.
But an empty shape `()` works and it matches torch.

Reply:
do we already have a test for this?

Reply:
There isn't.
I added a test in `test_tensor_data` but I dunno is it's right :s